app/routes/items/new.js

```diff
@@ -50,15 +50,15 @@ export default AuthorizeRoute.extend(GradeMixin, {
   },
 
   setupPrinterId(controller) {
-    let allAvailablePrinters = this.get(
-      "printerService"
-    ).allAvailablePrinters();
-    let user = this.get("session.loggedInUser");
-    if (user.get("printerId")) {
-      controller.set("selectedPrinterId", user.get("printerId"));
+    let defaultPrinter = this.get("printerService").getDefaultPrinter();
+    if (defaultPrinter) {
+      controller.set("selectedPrinterId", defaultPrinter.id);
     } else {
+      let allAvailablePrinters = this.get(
+        "printerService"
+      ).allAvailablePrinters();
       let firstPrinterId = allAvailablePrinters[0].id;
```

Comment:
This line would break if `allAvailablePrinters` is an empty array

Reply:
ohh !! thanks for pointing, will add guard clause.